### DISCUSSION TITLE
SolidOS new readme as dev landing page

### DISCUSSION THREAD

**Discussion by timea-solid** [solid/solidos#63] [Archive] [answered]
# Why would a person make use of the SolidOS github?
* if they are a user of a Pod and SolidOS 
  - send to user guide (the code does not help)
  -  if they have an issue send to making a git Issue 
* if they want to help improve/develop it: 
  - link to open issues
  - as a dev - keep reading (with anchors for architecture, deployment(including test server), and how to set up the code)
  -  as a UX designer - ?
  -  as a non-dev - ? 

# Proposed goals:
* talk in developer language but make it accessible for beginners too
* point to different docu locations (standard list of links in all docus)
* keep it simple
* a simple page that does not need maintenance often. Details should be edited in the according locations.

# Proposed table of content:
* Welcome message
  -  describe what the reader is expected to find here 
  -  links to other places if one is wrong here (standard link list)
  -  What is SolidOS
  -- both a user explanation and a technical explanation
  --  also point to what SolidOS is aspiring to be
  -- link to vision, mission, and goals for more details. -> to be decided where this best goes. (maybe on solidOs pod? or keep in readme)
  --  link to its history
* Architecture - not sure about this one
  - diagram
  - link to the other repositories part of it (solid-ui, solid-logic...)
* Want to help?
  - setting up the code (SolidOS locally)
  -- prerequisites
  -- how to
  -- link to FAQ in wiki for help
  -- link to gitter solidOS channel
  -- links to further docu
 *  How we work on SolidOS
   -- information about regular meetings, gitter, forum and so on...
   -- link to solidos pod for meeting times and meeting link
   -- maybe a buddy system?
   -- CoC
   
# remarks about what it should NOT contain:
* it should not contain any information about other projects except links to other project
* it should not explain other implementations that are not related to SolidOS. Those are to be documented in their own repos, at most contain links to them.

**Comment by timea-solid** [comment]
Remark - it was not clear to me that SolidOS installs an NSS too! -> which means one does not need NSS installed beforehand. This needs to be part of the docu too.

**Reply by angelo-v** [reply]
It's the other way arround: NSS comes with SolidOS by default

**Comment by bourgeoa** [comment]
NSS is a way among others to test solidOS.
solidOS should also be used as a front-end to CSS

**Comment by jeff-zucker** [comment]
CSS is also included now inside Data-Kitchen's node_modules.  My thought is that we should move it into its own solidos/workspace and have Data-Kitchen symlink to it there.   When you're looking at your local files in Data-Kitchen, that's a functioning CSS running.   That plus Data-Kitchen's ability to live update means that we can change either mashlib or CSS  and have the changes immediately reflected in the UI.

**Comment by jeff-zucker** [comment]
@theRealImy  asked about a tagline.  How about : SolidOS - semantic browsing, personal productivity, and collaboration tools for Solid.

**Reply by timea-solid** [reply]
I add Matthias' suggestion here too not to loose it: "The Default-App (Operating System) running on the Solid-Pod-Website functional as Databrowser"

**Comment by timea-solid** [comment, marked as answer]
See latest SolidOS readme: https://github.com/solid/solidos
